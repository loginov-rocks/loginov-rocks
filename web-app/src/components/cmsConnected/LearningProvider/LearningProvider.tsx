import * as React from 'react';
import { useEffect } from 'react';

import { ExternalLink } from 'components/shared/ExternalLink';
import { useLearningSectionContext } from 'contexts/LearningSectionContext';

interface Props {
  title: string;
  website?: string;
}

export const LearningProvider: React.FC<Props> = ({ title, website }) => {
  const { addProvider } = useLearningSectionContext();

  useEffect(() => {
    addProvider(title);
  }, []);

  if (website) {
    return <ExternalLink href={website} label="Link to Provider">{title}</ExternalLink>;
  }

  return <span title="Provider">{title}</span>;
};

LearningProvider.defaultProps = {
  website: undefined,
};
