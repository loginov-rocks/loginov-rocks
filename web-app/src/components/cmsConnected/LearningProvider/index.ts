import { LearningProvider } from './LearningProvider';

export { LearningProvider };
