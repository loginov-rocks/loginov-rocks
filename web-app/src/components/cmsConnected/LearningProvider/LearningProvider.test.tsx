/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { LearningProvider } from './LearningProvider';

it('matches snapshot', () => {
  const wrapper = shallow(<LearningProvider title="LinkedIn Learning" website="https://www.linkedin.com/learning/" />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot with optional fields', () => {
  const wrapper = shallow(<LearningProvider title="LinkedIn Learning" />);

  expect(wrapper).toMatchSnapshot();
});
