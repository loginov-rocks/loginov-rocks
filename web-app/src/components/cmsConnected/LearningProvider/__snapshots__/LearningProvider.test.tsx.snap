// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ExternalLink
  href="https://www.linkedin.com/learning/"
>
  LinkedIn Learning
</ExternalLink>
`;

exports[`matches snapshot with optional fields 1`] = `
<span>
  LinkedIn Learning
</span>
`;
